src/Components/Publishing/Layouts/NewsLayout.tsx

```diff
@@ -5,31 +5,51 @@ import { NewsSections } from "../News/NewsSections"
 
 interface Props {
   article: any
-  expanded?: boolean
+  isTruncated?: boolean
 }
 
 interface State {
-  expanded: boolean
+  isTruncated: boolean
+}
+
+interface NewsContainerProps {
+  isTruncated: boolean
 }
 
 export class NewsLayout extends Component<Props, State> {
   state = {
-    expanded: this.props.expanded || true,
+    isTruncated: this.props.isTruncated || true,
```

Comment:
I don’t know why this isn’t giving you a type error, but `this.props` is not what you expect it to be when assigning initial state statically, you have to do it from a `constructor` override instead.

Additionally, the assigned value is _always_ going to be `true`, because it’s essentially `(false || undefined) || true`. If the intention was to default to `true` than you’ll need to do `this.props.isTruncated == undefined ? true : this.props.isTruncated`

Reply:
Oh, actually [I’m wrong](https://www.typescriptlang.org/play/#src=class%20X%20%7B%0D%0A%09props%3A%20%7B%20foo%3A%20string%20%7D%0D%0A%0D%0A%09constructor(props)%20%7B%0D%0A%09%09this.props%20%3D%20props%09%0D%0A%09%7D%0D%0A%7D%0D%0A%0D%0Aclass%20Y%20extends%20X%20%7B%0D%0A%09state%20%3D%20%7B%0D%0A%09%09foo%3A%20this.props.foo%0D%0A%09%7D%0D%0A%7D%0D%0A%0D%0Aconst%20y%20%3D%20new%20Y(%7B%20foo%3A%20%22ok%22%20%7D)%0D%0Aconsole.log(y.state.foo))!

Looking at the generated code you can see that the `this.props` static state assignment is moved into the constructor, so as long as the superclass’ constructor is being used (which is the case because you don’t override it) then `this.props` should be available.

Having said that, I find it very unintuitive because it’s inside an object literal and thus in the lexical scope I’d expect `this` to refer to the class (if such a thing were even possible with ES classes). I wonder if this is according to the spec or just a side-effect of the compiler implementing static instance variable assignment by moving the code inside the constructor.

Reply:
Went a little deeper on this and asked for clarification on the proposal for class properties https://github.com/tc39/proposal-class-fields/issues/46#issuecomment-371440198.

Reply:
Using `this` in this way (even if it is transpiled out into the proper form) very much strikes me as a code smell! It's also unintuitive because I don't think this kind of behavior is seen often in other languages due to lack of rewrite step. Why not set state in constructor and keep the control flow universal?

Reply:
Ah yeah, this is not ideal I agree. Not sure why I did that - thanks for the catch!